plugins/CoreAdminHome/Commands/OptimizeArchiveTables.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\CoreAdminHome\Commands;
+
+use Piwik\Common;
+use Piwik\DataAccess\ArchiveTableCreator;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Plugin\ConsoleCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Administration command that optimizes archive tables (even if they use InnoDB).
+ */
+class OptimizeArchiveTables extends ConsoleCommand
+{
+    const ALL_TABLES_STRING = 'all';
+    const CURRENT_MONTH_STRING = 'now';
+
+    protected function configure()
+    {
+        $this->setName('core:optimize-archive-tables');
```

Comment:
I'm wondering if there's a different namespace that we could use instead of `core`? In the past we used too often `core` which doesn't say a lot. Many of them we already moved out of `core`. Not sure what to use here though. Maybe `database`? This would also make the context of `tables` clearer. Not everyone seeing this command will know what we mean by `archive-tables` so it's at least clearer that we're talking about a database. There might be better namespaces though, or `core`...